clash-prelude/src/Clash/Promoted/Nat.hs

```diff
@@ -115,15 +117,39 @@ instance ShowX (SNat n) where
 withSNat :: KnownNat n => (SNat n -> a) -> a
 withSNat f = f SNat
 
+-- | Same as 'snatToInteger', but doesn't take term arguments. Example usage:
+--
+-- >>> natToInteger @5
+-- 5
+natToInteger :: forall n . KnownNat n => Integer
```

Comment:
But we already have `GHC.TypeLits.natVal`... perhaps the comment should be that this is the ambiguous version of that